tests/integration/release/parallel_test_launcher.py

```diff
@@ -0,0 +1,218 @@
+        print(f"Failed to submit job: {script_name}\nError: {e}")
+        return None
+
+
+def is_job_running(job_id: str) -> bool:
+    """
+    Checks if a job with the given Slurm job ID is still running or pending.
+
+    Args:
+        job_id (str): Slurm job ID.
+
+    Returns:
+        bool: True if the job is running or pending; False if it has finished or if checking the job status fails.
+    """
+    try:
+        result = subprocess.run(
+            f"squeue --job {job_id}", shell=True, capture_output=True, text=True
+        )
+        return (
+            job_id in result.stdout
+        )  # The job ID is listed in squeue if it's running/pending
+    except subprocess.CalledProcessError as e:
+        print(f"Error checking job status: {e}")
+        return False
+
+
+def wait_for_all_jobs(job_ids: Sequence[str | None], poll_interval: int = 1800) -> None:
+    """
+    Waits for all jobs to complete, checking their status every `poll_interval` seconds.
+
+    Args:
+        job_ids (list): List of Slurm job IDs to monitor.
+        poll_interval (int): Time interval (in seconds) between status checks. Default is 1800 seconds (30 minutes).
+    """
+    remaining_jobs = set(
+        x for x in job_ids if x is not None
+    )  # Track jobs that are still running
+    while remaining_jobs:
+        print(f"Checking status of {len(remaining_jobs)} remaining jobs...")
+
+        completed_jobs = []
+        for job_id in remaining_jobs:
+            if not is_job_running(job_id):
+                print(f"Job {job_id} has finished.")
+                completed_jobs.append(job_id)
+
+        # Remove completed jobs from the remaining jobs set
+        remaining_jobs -= set(completed_jobs)
+
+        if remaining_jobs:
+            print(
+                f"{len(remaining_jobs)} jobs have not finished. Checking again in {poll_interval // 60} minutes..."
+            )
+            time.sleep(poll_interval)  # Wait for the specified interval before checking again
+
+
+def parse_outputs(output_dir: str, job_ids: list[str] | None = None) -> None:
+    """Parses the output files to see how many jobs failed, didn't complete, and succeeded
+    in the specified directory, prints this to screen, and writes a summary CSV with this info.
+
+    Args:
+        output_dir (str): Directory where Slurm output/error files are stored.
+        job_ids (list[str] | None): List of Slurm job IDs to check. If provided, will verify that all expected outputs are present.
+    """
+    job_info = []
+
+    output_files = glob.glob(f"{output_dir}/*.out")
+    if job_ids:
+        if len(output_files) < len(job_ids):
+            print(
+                f"Warning: Expected {len(job_ids)} job outputs, but found only {len(output_files)} in {output_dir}."
+            )
+            print(
+                "This may indicate that some outputs are still being written. Rerun parse_outputs after all outputs have been written"
```

Comment:
Is the expectation that the user will manually rerun this? But why? This gets called after `wait_for_all_jobs()`. It seems more like this should _error_ and would imply that someone went in and mucked around with things and so we can no longer trust results?